src/hooks/useModal.js

```diff
@@ -0,0 +1,24 @@
+import { useRef, useState } from "react";
+import useHandleModalClick from "./useHandleModalClick";
+
+export const useModal = () => {
+  const modalRef = useRef(null);
+  const [openModal, setOpenModal] = useState(false);
```

Comment:
P2:
boolean값인지 알 수 있는 이름이면 좋겠습니다.
```suggestion
  const [isOpen, setIsOpen] = useState(false);
```